### PR TITLE
Update fluentd 1.4.1

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature

/kind flake

**What this PR does / why we need it**:
- Update fluentd to 1.4.1.
- Update image README.md with the correct links to the image repository.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by monotek — CHANGES_REQUESTED** [review]
Update of version in makefile is missing.

**Comment by ialidzhikov** [comment]
Everything is fine with docker run gcr.io/fluentd-elasticsearch/fluentd:v2.5.0 - the fluentd version is 1.4.0. I updated the PR description.
```bash
$ docker run gcr.io/fluentd-elasticsearch/fluentd:v2.5.0
[...]
2019-04-01 15:26:52 +0000 [info]: gem 'fluentd' version '1.4.0'
[...]
```

File: cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml [hunk 1]
```diff
@@ -77,7 +77,7 @@ spec:
       serviceAccountName: fluentd-es
       containers:
       - name: fluentd-es
-        image: k8s.gcr.io/fluentd-elasticsearch:v2.5.0
+        image: gcr.io/fluentd-elasticsearch/fluentd:v2.5.1
```
Comment (coffeepac) [inline cluster/addons/fluentd-elasticsearch/fluentd-es-ds.yaml]: whoa, thanks for that!  I missed it during the switch over originally.

**Comment by coffeepac** [comment]
pushing the new image now, should be available in ~5 minutes.

**Comment by ialidzhikov** [comment]
Thank you!

**Pull request merged**